### PR TITLE
komo pizza babyyyy

### PR DESCRIPTION
**Pull request by gorlikariel**

### CODE REVIEW

File: client/src/components/Pizza.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  toppings: PizzaToppingData[];
+}
+
+interface StyledIconProps {
+  width?: string;
+  noGlow?: boolean;
+}
+
+export default function Pizza(props: PizzaProps) {
+  const { name, price, image, toppings, crust } = props;
```
Comment (nagauker) [inline client/src/components/Pizza.tsx:25]: open props in the props brackets

File: client/src/components/Pizza.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  image: string;
+  toppings: PizzaToppingData[];
+}
+
+interface StyledIconProps {
+  width?: string;
+  noGlow?: boolean;
+}
+
+export default function Pizza(props: PizzaProps) {
```
Comment (nagauker) [inline client/src/components/Pizza.tsx:24]: use es6

File: client/src/components/SelectTopping.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { ToppingData } from '../views/ToppingsView';
+import Icon from './Icon';
+import Label from './Label';
+
+interface SelectToppingProps {
+  addTopping: Function;
```
Comment (nagauker) [inline client/src/components/SelectTopping.tsx:8]: dont use 'Function' use () => .........

File: client/src/components/SelectTopping.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  removeTopping: Function;
```
Comment (nagauker) [inline client/src/components/SelectTopping.tsx:9]: same

File: client/src/components/SelectTopping.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  topping: ToppingData;
+}
+
+interface StarProps {
+  width?: string;
+}
+
+function SelectTopping(props: SelectToppingProps) {
+  const [count, setCount] = useState(0);
+  const { addTopping, removeTopping, topping } = props;
```
Comment (nagauker) [inline client/src/components/SelectTopping.tsx:19]: open the props in the brackets

File: client/src/index.css [hunk 6]
```diff
@@ -0,0 +1,13 @@
+body {
+  margin: 0;
+  background-image: url(https://png.pngtree.com/background/20210710/original/pngtree-delicious-pizza-background-material-picture-image_1050243.jpg);
+  background-size: cover;
+  font-family: 'Algerian';
+  -webkit-font-smoothing: antialiased;
```
Comment (nagauker) [inline client/src/index.css:6]: why?

File: client/src/index.css [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  -moz-osx-font-smoothing: grayscale;
```
Comment (nagauker) [inline client/src/index.css:7]: why?

File: client/src/views/BuildView.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  toppings: ToppingData[];
+}
+
+export default function BuildView() {
+  const [{ fetching, data }] = useQuery<ToppingsData>({ query: ToppingsQuery });
+  const [totalPrice, setTotalPrice] = useState(0);
+  // const [selectedToppings, setSelectedToppings] = useState([]);
+  const addTopping = (topping: ToppingData) => {
+    const { price, id } = topping;
+    console.log(id);
```
Comment (nagauker) [inline client/src/views/BuildView.tsx:26]: remove log

File: client/src/views/BuildView.tsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    setTotalPrice(totalPrice + price);
+  };
+
+  const removeTopping = (topping: ToppingData) => {
+    const { price, id } = topping;
+    console.log(id);
```
Comment (nagauker) [inline client/src/views/BuildView.tsx:32]: remove log

File: client/src/views/BuildView.tsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+  return (
+    <>
+      <Container>
+        <Label
+          fontSize='30px'
+          color='white'
+          text={`Total Price: ${totalPrice / 100}$`}
+        />
+        <OrderButton href='tel:0509012770'>!!ORDER NOW!!</OrderButton>
```
Comment (nagauker) [inline client/src/views/BuildView.tsx:44]: amazing!

File: client/src/views/BuildView.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    const { price, id } = topping;
+    console.log(id);
+    setTotalPrice(totalPrice - price);
+  };
+
+  return (
+    <>
+      <Container>
+        <Label
+          fontSize='30px'
```
Comment (nagauker) [inline client/src/views/BuildView.tsx:40]: px is not responsive

File: client/src/views/HomeView.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
+const HomeView = () => {
+  return (
+    <div
+      style={{
```
Comment (nagauker) [inline client/src/views/HomeView.tsx:4]: dont use inline style

File: client/src/views/HomeView.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+      }}
+    >
+      <h1>WELCOME TO KOMO PIZZA!</h1>
+      <span>
+        For orders call Ariel at -{' '}
+        <a
+          style={{
```
Comment (nagauker) [inline client/src/views/HomeView.tsx:16]: same

File: client/src/views/PizzasView.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  crust: string;
+  image: string;
+  pizza_toppings: PizzaToppingData[];
+}
+
+interface PizzasData {
+  pizzas: PizzaData[];
+}
+
+export default function PizzasView() {
```
Comment (nagauker) [inline client/src/views/PizzasView.tsx:19]: use es6

File: client/src/views/PizzasView.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+  const [{ fetching, data }] = useQuery<PizzasData>({ query: PizzasQuery });
+
+  return fetching ? null : (
```
Comment (nagauker) [inline client/src/views/PizzasView.tsx:22]: great chance to put loader

File: client/src/views/PizzasView.tsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+      })}
+    </PizzasContainer>
+  );
+}
+
+const PizzasContainer = styled.div`
+  display: grid;
+  justify-content: center;
+  grid-template-columns: repeat(1fr);
+  grid-gap: 40px;
```
Comment (nagauker) [inline client/src/views/PizzasView.tsx:45]: px is not responsive

File: client/src/views/PizzasView.tsx (hunk 15 above)
Comment (gorlikariel) [reply client/src/views/PizzasView.tsx:22]: The loading time is so fast that when I put a loader/skeleton it looks like a glitch on the screen... 

Screen Shot 2022-11-10 at 18 23 43

The options I can think about are leaving the page is it is, or using a loader and give it a timeout for a minimum display time of 1s, just so it doesn't look glitchy.

What do you think?

File: client/src/components/EditableTopping.tsx [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+`;
+
+const EditLabel = styled.span`
+  font-size: 1em;
+  font-weight: 600;
+  font-style: italic;
+`;
+
+const EditableTopping = (props: IEditableToppingProps) => {
+  const [mutationState, executeMutation] = useMutation(ToppingPriceMutation);
```
Comment (nagauker) [inline client/src/components/EditableTopping.tsx:87]: use significant names

File: client/src/components/EditableTopping.tsx [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+            <PriceDisplay>Price: {`${currentValue}$`}</PriceDisplay>
+          )}
+          <Edit
+            onClick={() => {
+              setEditMode(!editMode);
+            }}
+          >
+            <EditLabel>Edit</EditLabel>
+            <Icon width='1.5em' alt={'edit icon'} src={icons.edit} />
+            {edited ? (
```
Comment (nagauker) [inline client/src/components/EditableTopping.tsx:130]: use && instead of ? :

File: client/src/components/Icon.tsx [hunk 19]
```diff
@@ -0,0 +1,19 @@
+import styled from 'styled-components';
+
+interface StyledIconProps {
+  onClick?: React.MouseEventHandler;
+  width?: string;
+  alt: string;
+  src: string;
+}
+
+const StyledIcon = styled.img<StyledIconProps>`
```
Comment (nagauker) [inline client/src/components/Icon.tsx:10]: get only the width here

File: client/src/components/Layout.tsx [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+const Container = styled.div`
+  padding: 3.5em;
+  @media (max-width: 600px) {
+    padding: 2.2em;
+  }
+`;
+
+const Layout = () => {
+  return (
+    <div>
```
Comment (nagauker) [inline client/src/components/Layout.tsx:14]: div without anything... why not use a fragment?

File: client/src/components/Pizza.tsx [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+      rgb(31, 193, 27) 0px 0px 0px 6px, rgb(255, 217, 19) 0px 0px 0px 9px,
+      rgb(255, 156, 85) 0px 0px 0px 12px, rgb(255, 85, 85) 0px 0px 0px 15px;
+  }
+  @media (max-width: 700px) {
+    width: 19em;
+  }
+`;
+
+const Icon = styled.img<StyledIconProps>`
+  width: ${(props) => (props.width ? props.width : '5em')};
```
Comment (nagauker) [inline client/src/components/Pizza.tsx:68]: try: ??

File: client/src/components/Pizza.tsx [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+  box-shadow: ${(props) =>
```
Comment (nagauker) [inline client/src/components/Pizza.tsx:69]: I think it's cleaner to add the CSS property to the condition

File: client/src/components/SelectTopping.tsx [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    32% 57%,
+    2% 35%,
+    39% 35%
+  );
+  box-shadow: 0px 0px 9px 4px #8b4513;
+  min-height: 12em;
+  background-color: green;
+  display: flex;
+  flex-direction: column;
+  width: ${(props) => (props.width ? props.width : '35em')}}
```
Comment (nagauker) [inline client/src/components/SelectTopping.tsx:41]: try ??

File: client/src/views/ToppingsView.tsx [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+    grid-gap: 2vmax;
+  }
+`;
+
+const ToppingsView = () => {
+  const [{ fetching, data }] = useQuery<ToppingsData>({ query: ToppingsQuery });
+
+  return fetching ? (
+    <ToppingsContainer>
+      {[...new Array(12)].map(() => {
```
Comment (nagauker) [inline client/src/views/ToppingsView.tsx:33]: brilliant

File: client/src/components/Label.tsx [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+    95% {
+      filter: blur(0);
+      color: ${color || 'inherit'};
+      text-shadow: none;
+    }
+  }
+`;
+
+const StyledLabel = styled.span<LabelProps>`
+  font-size: ${(props) => `${props.fontSize ? props.fontSize : '1.22em'};`}
```
Comment (nagauker) [inline client/src/components/Label.tsx:32]: || ?

Comment (gorlikariel) [reply client/src/components/Label.tsx]: fixed

**Review by nagauker — APPROVED** [review]

**Review by nagauker — APPROVED** [review]

**Pull request merged**